Consumer/MessageConsumer.php

```diff
@@ -50,22 +52,45 @@ class MessageConsumer implements ConsumerInterface
     private $serializer;
 
     /**
+     * Producer used to resend message to the end of the queue
+     *
+     * @var Producer
+     */
+    private $resendProducer;
+
+    /**
+     * Producer used to send message to quarantine queue
+     *
+     * @var Producer
+     */
+    private $quarantineProducer;
+
+    /**
      * Constructor
      *
-     * @param MessageDispatcher   $messageDispatcher
-     * @param SerializerInterface $serializer
+     * @param MessageDispatcher   $messageDispatcher  Message dispatcher
+     * @param SerializerInterface $serializer         Message serializer
+     * @param Producer            $resendProducer     Producer used to resend message to the end of the queue
+     * @param Producer            $quarantineProducer Producer used to send message to quarantine queue
```

Comment:
It's worth considering a dead letter exchange as an alternative to a quarantine queue that exists as an out-of-the-box solution in AMQP protocol.